dev/src/types.ts

```diff
@@ -27,9 +30,62 @@ export interface ApiMapValue {
   [k: string]: google.firestore.v1.IValue;
 }
 
-// We don't have type information for the JavaScript GapicClient.
-// tslint:disable-next-line:no-any
-export type GapicClient = any;
+/**
+ * The subset of methods we use from FirestoreClient.
+ *
+ * We don't depend on the actual Gapic client to avoid loading the GAX stack at
+ * module initialization time.
+ */
+export interface GapicClient {
+  getProjectId(): Promise<string>;
+  beginTransaction(
+    request: api.IBeginTransactionRequest,
+    options?: CallOptions
+  ): Promise<[api.IBeginTransactionResponse, unknown, unknown]>;
+  commit(
+    request: api.ICommitRequest,
+    options?: CallOptions
+  ): Promise<[api.ICommitResponse, unknown, unknown]>;
+  rollback(
+    request: api.IRollbackRequest,
+    options?: CallOptions
+  ): Promise<[google.protobuf.IEmpty, unknown, unknown]>;
+  batchGetDocuments(
+    request?: api.IBatchGetDocumentsRequest,
+    options?: CallOptions
+  ): Duplex;
+  runQuery(request?: api.IRunQueryRequest, options?: CallOptions): Duplex;
+  listDocuments(
+    request: api.IListDocumentsRequest,
+    options?: CallOptions
+  ): Promise<[api.IDocument[], unknown, unknown]>;
+  listCollectionIds(
+    request: api.IListCollectionIdsRequest,
+    options?: CallOptions
+  ): Promise<[string[], unknown, unknown]>;
```

Comment:
FYI: I tried to use `Promise<[T, undefined, undefined]>` here, but the paginated methods and the non-paginated methods seem to disagree on return type (undefined vs null).

Reply:
Oh, this is something to be fixed in gax (later).